mgmtsystem/azure.py

```diff
@@ -440,6 +517,38 @@ def remove_blob_image(self, vm_vhd, container=None):
                        storage_key=self.storage_key,
                        container=container), True)
 
+    def remove_nics_by_search(self, nic_template):
```

Comment:
For these cleanup items, I added a new parameter a while back called ignore_error to run_script.  Generally speaking, you can set this to True so that you don't get exceptions.  A lot of tasks take a while to complete and I've stumbled into a lot of cases where by the time I tried to delete it it was already gone.  You're call, but something to think about.